src/NHibernate/NHibernateUtil.cs

```diff
@@ -399,6 +404,10 @@ public static bool IsInitialized(object proxy)
 			{
 				return !((INHibernateProxy)proxy).HibernateLazyInitializer.IsUninitialized;
 			}
+			else if (proxy is ILazyInitializedCollection)
```

Comment:
Maybe is it time to align implementation on `Initialize` for consistency and efficiency, by avoiding the next cast thanks to a variable declared in the test. If doing the change, do it for `IPersistentCollection` next `if` too.

Reply:
Good point, I adapted it.